Fix some flaky e2e tests

raethlein opened the pull request:
## Describe your changes

For some tests its difficult to reproduce them, so this is a best-effort change based on patterns we have seen in the past.

st_dialog_test.py:
- the sidebar snapshot test used the wrong dialog
- the dialog snapshot test was flaky due to different element focuses sometimes. Now, we enforce one specific focus
- the reopen-after-dismiss test was flaky -> increased the time between interactions

st_altair_chart_basic_select_test:
- fix regex escaping for `test_point_bar_chart_displays_selection_text`

st_date_input_test:
- wait after date is picked and checking the markdown (failed [here](https://github.com/streamlit/streamlit/actions/runs/9107613124))

custom_components/popular_components:
- fixed iframe locator that triggered error sometimes for some components that cause multiple iframes to render
- remove lottie as it hang indefinitely on my machine

## GitHub Issue Link (if applicable)

## Testing Plan

- E2E Tests: updated

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

raethlein committed:
Update test files

raethlein committed:
Fix character escaping

raethlein committed:
Fix large-width test

raethlein committed:
Add wait_for_app_run

raethlein committed:
Update snapshots

raethlein committed:
Add wait_for_app_run

kajarenc commented on e2e_playwright/st_dialog_test.py:
```diff
@@ -180,14 +184,18 @@ def test_largewidth_dialog_displays_correctly(
+    dialog.locator("div", has_text="Large-width Dialog").click()
+    submit_button = dialog.get_by_test_id("stButton")
+    expect(submit_button).to_be_visible()
+    submit_button.get_by_test_id("baseButton-secondary").hover()
```
`submit_button.hover()` doesn't work, and we need a more narrow locator here?

raethlein replied:
yep, I tried the other one first but apparently the effect happens on the button element itself!

raethlein committed:
Merge remote-tracking branch 'origin/develop' into tests/fix-flaky-dialog-tests

raethlein committed:
Add another wait_for_app_run

raethlein committed:
Try to make date-callback test more robust

raethlein committed:
Add tracing

raethlein committed:
Make dataframe selection more robust

raethlein committed:
Update trace names

raethlein committed:
Add click delay

raethlein committed:
Add debug log

raethlein committed:
Add browser name to trace

raethlein committed:
Change element to click on

raethlein committed:
Use input fill instead of type

raethlein committed:
Add altair-selection tests for forms, callback, and fragment (#8690)

## Describe your changes

Similar to the dataframe selection
tests, we want to test chart selections with callbacks and inside forms or
fragments explicitly.

Also fix two issues with the dataframe selection e2e tests.

raethlein committed:
Add re-try to opening the dialog

raethlein committed:
Increase timeout

raethlein committed:
Fix tests flakiness

raethlein committed:
Merge remote-tracking branch 'origin/develop' into tests/fix-flaky-dialog-tests

raethlein committed:
Remove tracing

raethlein committed:
Fix iframe locator error and remove lottie

raethlein committed:
Remove comment
